gef-ui/server/controllers/unissued-facilities/index.test.js

```diff
@@ -0,0 +1,383 @@
+import { add, format } from 'date-fns';
+import {
+  renderChangeFacilityPartial,
+  changeUnissuedFacility,
+  changeUnissuedFacilityPreview,
+  postChangeUnissuedFacility,
+  postChangeUnissuedFacilityPreview,
+} from './index';
+import api from '../../services/api';
+import MOCKS from '../mocks/index';
+
+jest.mock('../../services/api');
+
+describe('renderChangeFacilityPartial()', () => {
+  let mockRequest;
+  let mockFacilityResponse;
+
+  beforeEach(() => {
+    mockRequest = MOCKS.MockRequestUnissued();
+    mockFacilityResponse = MOCKS.MockFacilityResponseUnissued();
+
+    api.getApplication.mockResolvedValue({});
+    api.getFacility.mockResolvedValue(mockFacilityResponse);
+    api.updateFacility.mockResolvedValue({});
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('returns an object with expected parameters for changeUnissuedFacility', async () => {
+    mockRequest.query.status = 'change';
+
+    const result = await renderChangeFacilityPartial(mockRequest.params, mockRequest.query, true);
+
+    const expected = {
+      facilityType: 'CASH',
+      facilityName: 'Foundry4',
+      hasBeenIssued: true,
+      monthsOfCover: null,
+      shouldCoverStartOnSubmission: 'true',
+      issueDateDay: '5',
+      issueDateMonth: '1',
+      issueDateYear: '2022',
+      coverStartDateDay: '2',
+      coverStartDateMonth: '1',
+      coverStartDateYear: '2022',
+      coverEndDateDay: '2',
+      coverEndDateMonth: '1',
+      coverEndDateYear: '2030',
+      facilityTypeString: 'cash',
+      dealId: '123',
+      facilityId: 'xyz',
+      status: 'change',
+      change: true,
+    };
+
+    expect(result).toEqual(expected);
+  });
+
+  it('returns an object with expected parameters for changeUnissuedFacilityPreview', async () => {
+    mockRequest.query.status = 'change';
+    const result = await renderChangeFacilityPartial(mockRequest.params, mockRequest.query, false);
+
+    const expected = {
+      facilityType: 'CASH',
+      facilityName: 'Foundry4',
+      hasBeenIssued: true,
+      monthsOfCover: null,
+      shouldCoverStartOnSubmission: 'true',
+      issueDateDay: '5',
+      issueDateMonth: '1',
+      issueDateYear: '2022',
+      coverStartDateDay: '2',
+      coverStartDateMonth: '1',
+      coverStartDateYear: '2022',
+      coverEndDateDay: '2',
+      coverEndDateMonth: '1',
+      coverEndDateYear: '2030',
+      facilityTypeString: 'cash',
+      dealId: '123',
+      facilityId: 'xyz',
+      status: 'change',
+      change: false,
+    };
```

Comment:
This seems to be duplicated in 2 places(?) are we able to use a reusable mock for this, keep it DRY?

Reply:
done